nowcasting_dataset/data_sources/nwp/nwp_data_source.py

```diff
@@ -144,26 +156,43 @@ def sample_period_minutes(self) -> int:
         return 60
 
 
-def open_nwp(zarr_path: str, consolidated: bool) -> xr.Dataset:
+def open_nwp(zarr_path: str, consolidated: bool) -> xr.DataArray:
     """
     Open The NWP data
 
     Args:
         zarr_path: zarr_path must start with 'gs://' if it's on GCP.
-        consolidated: consolidate the zarr file?
-
-    Returns: nwp data
+        consolidated: Is the Zarr metadata consolidated?
 
+    Returns: NWP data.
     """
     _LOG.debug("Opening NWP data: %s", zarr_path)
     utils.set_fsspec_for_multiprocess()
     nwp = xr.open_dataset(
         zarr_path, engine="zarr", consolidated=consolidated, mode="r", chunks=None
     )
 
-    # Sanity check.
-    # TODO: Replace this with
-    # pandas.core.indexes.base._is_strictly_monotonic_increasing()
-    assert utils.is_monotonically_increasing(nwp.init_time.astype(int))
-    assert utils.is_unique(nwp.init_time)
-    return nwp
+    ukv = nwp["UKV"]
+
+    # Sanity checks.
+    # If there are any duplicated init_times then drop the duplicated init_times:
+    init_time = pd.DatetimeIndex(ukv["init_time"])
+    if not init_time.is_unique:
+        _LOG.warning("NWP Zarr has duplicated init_times.  Fixing...")
```

Comment:
good warning

Reply:
I like it where it says how many duplicates were dropped too. nice to have thought

Reply:
Good idea - I'll add that!  Thanks!